### PR TITLE
#128 signup login frontend validation

### PR DESCRIPTION
**Pull request by Pooja-Codes**
- Added frontend validation on the signup and login page.
- Updated UI to show validation message
- Added package vuelidate
- Validation rules can be updated later according to team suggestions. The minor flow bug should be fixed with API integration.

### CODE REVIEW

File: src/frontend/shortlist/src/components/layout/signup_login/Login.vue [hunk 1]
```diff
@@ -1,31 +1,50 @@
 <script>
 import Logo from "./Logo.vue";
 import { userLoginStore } from "../../../states/userLogin";
+import useVuelidate from "@vuelidate/core";
```
Comment (simaoalicechen) [inline src/frontend/shortlist/src/components/layout/signup_login/Login.vue:4]: Really good way to do this.

File: src/frontend/shortlist/src/components/layout/signup_login/Signup.vue [hunk 2]
```diff
@@ -1,38 +1,104 @@
...
+import {
+  required,
+  email,
+  minLength,
+  maxLength,
+  sameAs,
+} from "@vuelidate/validators";
+
+export function validName(name) {
+  let validNamePattern = new RegExp("^[a-zA-Z]+(?:[-'\\s][a-zA-Z]+)*$");
```
Comment (simaoalicechen) [inline src/frontend/shortlist/src/components/layout/signup_login/Signup.vue:14]: Good testing function.

File: src/frontend/shortlist/src/components/layout/signup_login/Login.vue (hunk 1 above)
Comment (Pooja-Codes) [reply src/frontend/shortlist/src/components/layout/signup_login/Login.vue:4]: Thank You!

File: src/frontend/shortlist/src/components/layout/signup_login/Signup.vue [hunk 3]
```diff
@@ -51,34 +117,106 @@ export default {
...
+
+      <button
+        type="submit"
+        :disabled="v$.form.$invalid"
+        class="btn btn-outline-dark"
+        @click="submitSignupForm"
+      >
+        Sign up
+      </button>
       <p class="instructions" id="small">Have an account already?</p>
```
Comment (simaoalicechen) [inline src/frontend/shortlist/src/components/layout/signup_login/Signup.vue:220]: I realize that you have LoginWithSSO in Login, but does not have SignupWithSSO. Maybe add them or delete both of them?

**Review by simaoalicechen — CHANGES_REQUESTED** [review]

File: src/frontend/shortlist/src/components/layout/signup_login/Signup.vue (hunk 3 above)
Comment (Pooja-Codes) [reply src/frontend/shortlist/src/components/layout/signup_login/Signup.vue:220]: The user can not signup with google due to an ID upload requirement. However, they can log in with google if they are verified. Feel free to discuss about application flow doubts/suggestions later in the meeting.

Comment (simaoalicechen) [reply src/frontend/shortlist/src/components/layout/signup_login/Signup.vue:220]: Nice.  One thing to communicate with the backend is that if we use the LoginWithSSO, when the user login with a Google account (an email account I assume), email is a required variable then.

Comment (Pooja-Codes) [reply src/frontend/shortlist/src/components/layout/signup_login/Signup.vue:220]: Email is the required variable, you are right. Also, there is NO username according to the product owner-note for theAPI.

**Comment by waltage** [comment]
LGTM; get alice's review ironed out and we're good

**Review by waltage — APPROVED** [review]

**Review by simaoalicechen — APPROVED** [review]

**Review by yalexublue — APPROVED** [review]
LGTM

**Pull request merged**